packages/blocks/src/api/factory.js

```diff
@@ -91,15 +91,21 @@ export function createBlock( name, attributes = {}, innerBlocks = [] ) {
 export function cloneBlock( block, mergeAttributes = {}, newInnerBlocks ) {
 	const clientId = uuid();
 
+	// Check for innerBlocks on block before attempting
+	// to map them. There are edge cases where they can
+	// be `undefined`
+	if ( ! newInnerBlocks && block.innerBlocks ) {
```

Comment:
What are these edge cases?

Reply:
Could be a signal of a missing `createBlock` call or something like that?

Reply:
I need to trace the instances where `blocks.innerBlocks` was `undefined`. All my calls are using `createBlock`/`cloneBlock` so I don't _think_ it was that but...

Reply:
I think it worths doing this checking, maybe just changing/updating the comment?

Reply:
I think I've solved the mystery! `innerBlocks` were truly undefined for all previews of reusable blocks called from `inserter/menu.js` and it caused bugs described above (and in one more review thread in this PR). 

This has happened because we've called `cloneBlocks` with the "block" object and new attributes. However, that "block" object was an **object describing block**, but not a block itself. This is the object we've been using:

https://github.com/WordPress/gutenberg/blob/823ba362fa3ca0016ef5f690e8f267420b994697/packages/block-editor/src/store/selectors.js#L1233-L1245

It shares some properties with a true block object (one created by `createBlock`) - `id` and `name` which made it kinda work with `cloneBlock` but it is definitely missing `innerBlocks` for example.

Knowing now that we don't actually have any block ready for a preview in the inserter, I have replaced the `cloneBlock` with a `createBlock` and that solved all issues around `innerBlocks`.

> Could be a signal of a missing createBlock call or something like that?

Spot on but it was a journey to discover why exactly 😀

Reply:
Fixed in https://github.com/WordPress/gutenberg/pull/16033/commits/2b8f1fb6527278eff3c6e0dcac78d28cbbbd7619

Reply:
Nice sleuthing. This is what I had in mind to check this morning so thanks for sorting it.